src/components/screens/MainScreen.tsx

```diff
@@ -58,6 +60,15 @@ const bottomBlockWrapper = classnames(
 )
 
 export default function () {
+  const [, setLocation] = useLocation()
+
+  useEffect(() => {
+    const urlSearchParams = new URLSearchParams(window.location.search)
+
+    if (urlSearchParams.has('token'))
+      setLocation(`/token/${urlSearchParams.get('token')}`)
+  }, [setLocation])
```

Comment:
Do we really need this?

Reply:
We have approved the link format with a simple query parameter `/?token=<token>`